dns_cache/manager.go

```diff
@@ -0,0 +1,62 @@
+package dns_cache
+
+import (
+	"context"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/TykTechnologies/tyk/log"
+)
+
+var (
+	logger = log.Get().WithField("prefix", "dns-cache")
+)
+
+type dialContextFunc func(ctx context.Context, network, address string) (net.Conn, error)
+
+type IDnsCacheManager interface {
+	InitDNSCaching(ttl, checkInterval time.Duration)
+	WrapDialer(dialer *net.Dialer) dialContextFunc
+}
+
+type DnsCacheManager struct {
+	DnsCache *DnsCacheStorage
+}
+
+func NewDnsCacheManager() *DnsCacheManager {
+	return &DnsCacheManager{nil}
```

Comment:
not clear why we do not prepare here usable `DnsCacheManager` instantiating `DnsCache` field instead of nil, that I guess purpose of these constructor-like functions